.setup/build/dev/Dockerfile

```diff
@@ -0,0 +1,10 @@
+FROM golang:1.18
+
+WORKDIR /app
+
+COPY . .
+
+RUN go install github.com/githubnemo/CompileDaemon@latest && \
+    go mod download
+
+ENTRYPOINT ["CompileDaemon", "-directory=./", "-build=go build -o main main.go", "-command=./main"] 
```

Comment:
CompileDaemon really improves coding time, not needing to `go run` by hand, but I understood your point when i think about a sample app. My decision here would be to have a section more detailed about this point at the README, what you think?
About the docker, I still don't know the dev/prod stack of riot, so while it might be different of what we going to use, I added docker because I wanted to show the demo presentation deployed in a Kubernetes environment (this work was improved at branch https://github.com/lucasMesquitaBorges/riot-dev-api-proxy/tree/setup-k8s).
The Kubernetes cluster is only temporary, while would be nice to show in a demo, it won't be running for the long term, so as a sample project the idea should be discarded.